src/main/java/core/basesyntax/Lottery.java

```diff
@@ -0,0 +1,16 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery {
+    private static final int NUMBER_COUNT = 101;
+    private Random random = new Random();
+    private Ball ball = new Ball();
+    private ColorSupplier color = new ColorSupplier();
+
+    public Ball getRandomBall() {
+        ball.setNumber(random.nextInt(NUMBER_COUNT));
+        ball.setColor(color.getRandomColor());
+        return ball;

```

Comment:
I would recommend here to refactor this part of code. First of all, here we are using the same instance of `ball` per each `getRandomBall() ` method call. which potentially can cause some errors in the future. Also, generally it is not recommended to have partially-stated objects, means that we create our instance without meaningful data, and only at some point are setting `color` and `number`. So, I would recommend to create `new Ball(number, color)` constructor and return each time new ball.